storage-api/src/java/org/apache/hive/common/util/BloomKFilter.java

```diff
@@ -36,7 +44,7 @@
  *
  * This implementation has much lesser L1 data cache misses than {@link BloomFilter}.
```

Comment:
I am afraid changing the logic of the implementation will also affect the cache effectiveness/behaviour -- would it make sense to add a JMH test here as well?
We could even save the bench-results in a log to keep track of whats happening

Reply:
Actually, I was having the same thoughts :) FYI: I did some JMH benhcmarks (outside Hive) some time ago. I am adapting them slightly and I will share some JMH results later today.